src/teleop/jetson/jetson_teleop.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+# Node for teleop-related callback functions
+
+from math import copysign
+from enum import IntEnum
+import rospy as ros
+from sensor_msgs.msg import Joy, JointState
+from geometry_msgs.msg import Twist
+from mrover.msg import Chassis, RAOpenLoopCmd, HandCmd
+
+
+def quadratic(val):
+    return copysign(val**2, val)
+
+
+def deadzone(magnitude, threshold):
+    temp_mag = abs(magnitude)
+    if temp_mag <= threshold:
+        temp_mag = 0
+    else:
+        temp_mag = (temp_mag - threshold) / (1 - threshold)
+
+    return copysign(temp_mag, magnitude)
+
+
+class Drive:
+    def __init__(self, joystick_mappings, drive_config, track_radius, wheel_radius):
+        self.joystick_mappings = joystick_mappings
+        self.drive_config = drive_config
+
+        # Constants for diff drive
+        self.TRACK_RADIUS = track_radius  # meter
+        self.WHEEL_RADIUS = wheel_radius  # meter
+        self.drive_vel_pub = ros.Publisher("/drive_cmd_wheels", Chassis, queue_size=100)
+
+    # TODO: Reimplement Dampen Switch
+    def teleop_drive_callback(self, msg):
+
+        # teleop_twist_joy angular message is maxed at 0.5 regardless of
+        # turbo mode, multiply by 2 to make range [-1,1]
+        v = msg.linear.x
+        omega = msg.angular.z * 2
+
+        # Transform into L & R wheel angular velocities using diff drive kinematics
+        omega_l = (v - omega * self.TRACK_RADIUS / 2.0) / self.WHEEL_RADIUS
+        omega_r = (v + omega * self.TRACK_RADIUS / 2.0) / self.WHEEL_RADIUS
+
+        command = Chassis()
+        command.omega_l = omega_l
+        command.omega_r = omega_r
+
+        self.drive_vel_pub.publish(command)
+
+
+class ArmControl:
+    def __init__(self, xbox_mappings, ra_config):
+        self.xbox_mappings = xbox_mappings
+        self.ra_config = ra_config
+
+        # RA Joint Publishers
+        self.joint_a_pub = ros.Publisher("/ra/open_loop/joint_a", JointState, queue_size=100)
+        self.joint_b_pub = ros.Publisher("/ra/open_loop/joint_b", JointState, queue_size=100)
+        self.joint_c_pub = ros.Publisher("/ra/open_loop/joint_c", JointState, queue_size=100)
+        self.joint_d_pub = ros.Publisher("/ra/open_loop/joint_d", JointState, queue_size=100)
+        self.joint_e_pub = ros.Publisher("/ra/open_loop/joint_e", JointState, queue_size=100)
+        self.joint_f_pub = ros.Publisher("/ra/open_loop/joint_f", JointState, queue_size=100)
+
+        # Hand Publishers
+        self.finger_pub = ros.Publisher("/hand/open_loop/finger", JointState, queue_size=100)
+        self.grip_pub = ros.Publisher("/hand/open_loop/grip", JointState, queue_size=100)
+
+    def ra_control_callback(self, msg):
+
+        joint_a = JointState()
```

Comment:
I feel like an enum can be used here (for joint letter) and some form of dictionary can be used to map that to the actual joint object.

Logic seems duplicate currently.

Reply:
https://docs.python.org/3/library/enum.html

Reply:
I'm somewhat confused by this comment, what would the enum actually contain? Would it be like joint_a=1, joint_b=2 etc. and then have the value of the enum be the key to the dict? I'm unsure of what the actual usage/benefit here would be.

Reply:
It could probably be like enum RAJoints and then the different enums could be like joint_a, joint_b. joint_c, etc. And then you can make a dict called control_by_joint where u do something like:
control_by_joint[JointF] = msg.axes[self.xbox_mappings["right_bumper"]] - msg.axes[self.xbox_mappings["left_bumper"]];

Then you can iterate through the RAJoints to avoid code duplication by doing for ra_joint in RAJoints:
            self.ra_config[ra_joint]["multiplier"]
            * (control_by_joint[ra_joint])

Reply:
Ok the control_by_joint thing doesn't work straight from my example since xbox_mappings is constantly changing, but you kinda get my point

Reply:
Yeah I think some way to reduce the code duplication here would be nice. Could use some sort of dictionary or just a function that takes the joint name and xbox mapping